### PR TITLE
Test get claim params

### LINKED ISSUE #615: test: `/boost/get_claim_params?boost_id=${id}&addr=${addr}`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by estherbreath**
## Test: /boost/ get_claim_params

- it should fetch and return data for a valid  id and address

-  it should handle fetch errors in unexpected format

-  it should handle fetch with No Response 

resolved #615

### CODE REVIEW

File: tests/services/apiService.test.js [hunk 1]
```diff
@@ -40,3 +40,54 @@ describe("fetchQuestCategoryData function", () => {
...
+  }  
+  );
+  
+  it("should fetch and return data for a valid  id and address", async () => {
+         const mockData = ["string", "string"]; 
+         fetch.mockResolvedValueOnce({
+          json: () => Promise.resolve(mockData),
+        });
+  
+         const result = await getQuestBoostClaimParams ("string", "address");
```
Comment (ayushtom) [inline tests/services/apiService.test.js:57]: i dont think this is a valid address and id. Please check the api response and then decide the mockdata and errors

File: tests/services/apiService.test.js [hunk 2]
```diff
@@ -40,3 +40,54 @@ describe("fetchQuestCategoryData function", () => {
...
+  
+     const result = await getQuestBoostClaimParams ("456", "566");
+     expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/boost/get_claim_params?boost_id=456&addr=566`
+     );
+     expect(result).toEqual(mockResponse);
+  })
+  
+  
+  it("should handle fetch with No Response ", async () => {
```
Comment (ayushtom) [inline tests/services/apiService.test.js:79]: should be renamed to - should handle sending non-formatted data

File: tests/services/apiService.test.js [hunk 3]
```diff
@@ -40,3 +40,54 @@ describe("fetchQuestCategoryData function", () => {
...
+  });
+  
+  it("should handle fetch errrors in unexpected format", async () => {
+      const mockResponse = "Boost with id 456 not found";
+  
+      fetch.mockResolvedValueOnce({
+       json: () => Promise.resolve(mockResponse),
+     });
+  
+     const result = await getQuestBoostClaimParams ("456", "566");
```
Comment (ayushtom) [inline tests/services/apiService.test.js:71]: i think the `id` field is a number here.

**Comment by ayushtom** [comment]
@estherbreath i will be assigning someone else to finish this PR today. If you can then please commit the changes here so i can review and merge or else we give it to someone who can close the issue

**Comment by Oshioke-Salaki** [comment]
can i have this?

**Comment by Dprof-in-tech** [comment]
please can i take this

**Comment by 0xOlivanode** [comment]
Can I work on this ?

**Comment by ayushtom** [comment]
closing this pr

**Pull request closed without merging**